File: wrappers/java/integration_tests/src/com/lighting/huestream/tests/integration/helpers/IBridgeWrapper.java
package com.lighting.huestream.tests.integration.helpers;

import java.util.List;

public interface IBridgeWrapper {
    interface ILightCoordinate {
        Integer getLightId();
        Double getX();
        Double getY();
    }
    interface ILightID {
        Integer getLightId();
        String getMac();
    }

    String getLightRBGColor(ILightID lightId);

    Integer getEntertainmentGroupId();

    void removeEntertainmentGroups();

    List<ILightID> getLLCLightsIDs();

    void includeLightsIntoGroup(final List<ILightID> lights, final Integer targetGroupId);

    void setLightsCoordinates(final Integer groupId, List<ILightCoordinate> lightCoordinates);

    void cleanUpUser();
}

File: edk/wrappers/csharp/huestream_csharp_integrationtests/helpers/IBridgeWrapper.cs
using System;
using System.Collections.Generic;

namespace huestream_tests
{
    public interface ILightCoordinate
    {
        int GetLightId();
        double GetX();
        double GetY();
    }

    public interface ILightID
    {
        int GetLightId();
        String GetMac();
    }

    public interface IBridgeWrapper
    {
        void RemoveEntertainmentGroups();

        int GetEntertainmentGroupId();

        String GetLightRGBColor(ILightID lightId);

        List<ILightID> GetLLCLightsIDs();

        void IncludeLightsIntoGroup(List<ILightID> lights, int targetGroupId);

        void SetLightsCoordinates(int groupId, List<ILightCoordinate> lightCoordinates);

        void CleanUpUser();
    }
}
